module/certificateTemplates/helper.js

```diff
@@ -0,0 +1,179 @@
+            if ( !data.solutionId ) {
+                delete data.solutionId;
+            }
+            //  If programId value passed as empty string. 
+            if ( !data.programId ) {
+                delete data.programId;
+            }
+            let updateObject = {
+                "$set" : data
+              };
+            let certificateTemplateUpdated = 
+            await database.models.certificateTemplates.findOneAndUpdate(
+                {_id: templateId},
+                updateObject
+            );
+            return resolve({
+                message : constants.apiResponses.CERTIFICATE_TEMPLATE_UPDATED,
+                data : {
+                  id : certificateTemplateUpdated._id
+                }
+              });
+            
+        } catch (error) {
+            return reject(error);
+        }
+    });    
+  }
+
+  /**
+   * upload certificate template.
+   * @method uploadToCloud
+   * @name uploadToCloud
+   * @param {Object} fileData - file to upload.
+   * @param {String} templateId - templateId.
+   * @param {String} userId - user Id.
+   * @returns {JSON} Uploaded certificate template details. 
+  */
+  
+  static uploadToCloud(fileData, templateId, userId = "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+          const now = new Date();
+          const date = now.getFullYear() + "-"+ now.getMonth() + "-" + now.getDate() + "-" + now.getTime();
+          const fileName = userId + "_" + date + ".svg"; 
+          const requestData = {
+            "templates": {
+              "files": [fileName]
+            }
+          };
+
+          let signedUrl =
+          await filesHelpers.preSignedUrls(
+              requestData,
+              constants.common.CERTIFICATE,
+              userId,
+              templateId
+          );
+          
+          //  upload file using signed Url
+          if (signedUrl.data && 
+            Object.keys(signedUrl.data).length > 0 &&
+            signedUrl.data.templates &&
+            signedUrl.data.templates.files.length > 0 &&
+            signedUrl.data.templates.files[0].url &&
+            signedUrl.data.templates.files[0].url !== ""
+          ) {
+             
+            let fileUploadUrl = signedUrl.data.templates.files[0].url;
+            let file = fileData.file.data;
+           
+            try { 
+                await request({
+                  url: fileUploadUrl,
+                  method: 'put',
+                  headers: {
+                      "x-ms-blob-type" : "BlockBlob",
+                      "Content-Type": "multipart/form-data"
+                    },
+                  body: file
+                })
+                //  Update certificate template url in certificateTemplates collection
+                let updateCertificateTemplate = await this.update(
+                  templateId,
+                  { 
+                    templateUrl : signedUrl.data.templates.files[0].payload.sourcePath
```

Comment:
@VISHNUDAS-tunerlabs please add in comment a sample templateUrl how will it look like

Reply:
@aks30  added